lib/rules/no-duplicate-case.js

```diff
@@ -0,0 +1,57 @@
+/**
+ * @fileoverview Rule to disallow a duplicate case label.
+ * @author Dieter Oberkofler
+ * @copyright 2015 Dieter Oberkofler. All rights reserved.
+ */
+
+"use strict";
+
+//------------------------------------------------------------------------------
+// Rule Definition
+//------------------------------------------------------------------------------
+
+module.exports = function(context) {
+
+    /**
+     * Get a hash value for the test
+     * @param {ASTNode} test The "test" node.
+     * @returns {string} A hash value for the test.
+     * @private
+     */
+    function getHash(test) {
+        if (test.type === "Literal") {
+            return test.type + typeof test.value + test.value;
+        } else if (test.type === "Identifier") {
+            return test.type + typeof test.name + test.name;
+        }
+    }
+
+    return {
+
+        "SwitchStatement": function(node) {
+
+            var caseCount = node.cases.length,
+                hashTable = {},
+                caseNode,
+                hashValue,
+                i;
+
+            for (i = 0; i < caseCount - 1; i++) {
```

Comment:
The main reason is performance. ESLint is already doing a traversal, so you shouldn't also be doing a traversal.